src/v/cluster/topics_frontend.cc

```diff
@@ -195,10 +198,46 @@ ss::future<std::vector<topic_result>> topics_frontend::update_topic_properties(
       });
 }
 
+ss::future<std::error_code> topics_frontend::do_update_data_policy(
+  topic_properties_update& update, model::timeout_clock::time_point timeout) {
+    // Alter config rewrites all properties, so we need to delete dp if op is
+    // set and vaule is nullopt
+    bool is_convert_from_set_to_remove = false;
+    if (
+      update.properties.data_policy.op == incremental_update_operation::set
+      && !update.properties.data_policy.value.has_value()) {
+        update.properties.data_policy.op = incremental_update_operation::remove;
+        is_convert_from_set_to_remove = true;
+    }
+
+    switch (update.properties.data_policy.op) {
+    case incremental_update_operation::set:
```

Comment:
I think `ALTER_CONFIG` set operation should override current data policy. this is how it works for other topic properties

Reply:
https://github.com/vectorizedio/redpanda/pull/2167#discussion_r698976991

Reply:
I mean we do not want to use topic properties logic for data_policy? Or I need to  do the same things (rewrite if it exists, or return ok for delete if it does not exist) like for another properties?

I am a little bit confused

Reply:
I think if we want to used the `ALTER_CONFIG` to set the data policy it have to behave in the same way as for other properties.